packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -56,308 +57,358 @@ const extractTextPluginOptions = shouldUseRelativeAssetPaths
     { publicPath: Array(cssFilename.split('/').length).join('../') }
   : {};
 
+const getBabelPresetTarget = target => {
+  let babelPresetTarget;
+  if (target === 'module') {
+    // build an es6 compatible bundle
+    // https://philipwalton.com/articles/deploying-es2015-code-in-production-today/
+    // target all browsers which support the "module" attribute on script
+    babelPresetTarget = {
+      browsers: [
+        'Chrome >= 60',
+        'Safari >= 10.1',
+        'iOS >= 10.3',
+        'Firefox >= 54',
+        'Edge >= 15',
+      ],
+    };
+  } else {
+    babelPresetTarget = {
+      // React parses on ie 9, so we should too
+      ie: 9,
+      // We currently minify with uglify
+      // Remove after https://github.com/mishoo/UglifyJS2/issues/448
+      uglify: true,
+    };
+  }
+  return babelPresetTarget;
+};
+
+const getBabelPreset = (babelPresetObj, presetTarget) => {
+  const clonedBabelPresetObj = Object.create(babelPresetObj);
+  clonedBabelPresetObj.presets[0][1].targets = presetTarget;
```

Comment:
changes in https://github.com/facebookincubator/create-react-app/pull/3103 would make this less hacky IMO

Reply:
Isn't babel 7.0 [babelrc.js](https://github.com/babel/babel/issues/4630) a solution to this?

Reply:
@diligiant Not really, because as I mentioned below, both `.babelrc`and `babelrc.js` are required once and then cached. So can't switch things in them based on env variables for eg.

Reply:
@neeharv you mean there is no way you could offload the configuration to babelrc.js, invoking it yourself, in order to “preserve” some kind of consistency in regard to how Babel will be usually configured from now on?
(Not sure I make myself understandable ;)